tardieu/mcad
# Add build info to logs

This PR adds a log line with the build version and build date of the MCAD.
Knowing the build version and build date is useful when debugging historical logs and correlating them with the actual code that generated them.

build version format:
```
<tag>[-<sha>][-dirty]
```

`<tag>`: The name of the most recent tag (stripping the prefix 'v')
`<sha>`: If HEAD is not pointed to a tag, then the hash of HEAD is added.
`dirty`: If the repository contains any changes, "-dirty" is added.


# Verification steps
Build and run:
```
make build && ./bin/manager
```

The first log line should look like:
```
2023-11-20T15:20:43+02:00       INFO    setup   Build info      {"mcadVersion": "2.0.0-413e866-dirty", "date": "2023-11-20 15:20"}
```
